torch/csrc/jit/tensorexpr/reduction.h

```diff
@@ -162,9 +197,36 @@ class TORCH_API ReduceOp : public ExprNode<ReduceOp> {
     return reduce_args_;
   }
 
+  void setAccBuf(BufHandle acc_buf) {
+    acc_buf_ = acc_buf.node();
+  }
+  BufPtr getAccBuf() {
+    return acc_buf_;
+  }
+
+  void setResultBuf(BufHandle buf) {
+    result_buf_ = buf.node();
+  }
+  BufPtr getResultBuf() {
+    return result_buf_;
+  }
+
+  void setRiOperand(ExprHandle ri_operand) {
+    ri_operand_ = ri_operand.node();
+  }
+  ExprPtr getRiOperand() {
+    return ri_operand_;
+  }
+
  private:
+  // body_ = reducer_->interaction_(result_buf_, ri_operand_)
   ExprPtr body_;
   std::vector<VarPtr> reduce_args_;
+
+  BufPtr result_buf_;
+  BufPtr acc_buf_;
+  ExprPtr ri_operand_;
+
```

Comment:
Tbh, I don't think this is the right direction. 

We need to keep core concepts minimal and this PR hangs lots of things on ReduceOp. Please note that result_buf semantically existed before this change too and yet we didn't have to put it into this class - because it can be deduced from the parent Stmt (it is expected to be the store target and the reduce-op would be the store value). Now, with this change we introduce redundancy: result_buf needs to be set in reduce-op, but it still exists in the store target. What happens if they diverge? How do we prevent this? The answer is to not have them in two places like we've had before. And I suspect that we don't need the other two new fields in this class either.

Reply:
The motivation to add the three fields is to rebuild the reduction as we need to use a high precision accumulation buffer for low precision.

The [rebuild ](https://github.com/pytorch/pytorch/pull/84402/files#diff-fac6725328dc01e235944c7afc9f29c804488973c02c25ecd93d562884d959b3R27-R35)happens at the Tensor creation stage while the `store` has not been created. Besides that, it is hard to get `result_buf` from `ReduceOp`. The `result_buf` is the input of [`ReduceInteraction`](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/tensorexpr/reduction.h#L110) while the `ReduceInteraction` semantic is black-box as the caller could customize it at will. 

> The answer is to not have them in two places like we've had before.

Currently, we only use them once to rebuild the reduction. @ZolotukhinM, do you prefer we do not add the three field but just pass them as the parameters of `Tensor::constructStmt`?